src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1415,6 +1419,20 @@ trait RuntimeCosting extends SigmaLibrary with DataCosting with Slicing { IR: Ev
           withDefaultSize(res, cost)
       }
 
+      case XorOf(input) => input match {
+        case ConcreteCollection(items, tpe) =>
+          val itemsC = items.map(item => eval(adaptSigmaBoolean(item)))
```

Comment:
@greenhat why `adaptSigmaBoolean` is necessary. BoolValue is not related to SigmaBoolean anymore.

Reply:
Fixed.